### DISCUSSION TITLE
Addition to data product: what data to publish

### DISCUSSION THREAD

**Discussion by HJTP** [datamesh-architecture/datamesh-architecture.com#18] [Ideas]
Thanks for the website and its great explanation of data mesh! I have a proposal for an addition: what data would typically be included in a data product, and also which data shouldn't.

To be more precise, teams would be strongly urged to publish all their operational data, to enable other teams to use it. If they don't, this operational data will/should not be available anywhere, because this domain is the only proper source. On the other hand, it is undesirable for teams to republish data in unchanged form that they themselves got from another domain, because the "golden source" of this data apparently lies elsewhere. It goes without saying that if a team performs any transformation on the data, i.e., enriching it, filtering it, anything, the data can be republished because this team will be the golden source of this transformed data.

What do you think?

**Comment by jochenchrist** [comment]
HJTP,

thanks for your proposal.

Data that is created and owned by a domain are prime candidates, and the domain team should be encouraged to publish them in an appropriate, cleaned and managed form.

For source-aligned domains, I mostly would argue to include reference IDs. 
It is OK to include other domain's data, if the data was transformed, is the basic for business decisions or the exact state of the data at a processing time was relevant. In fact, these are cases, when the processing domain takes ownership for these data based on business cases. 

Aggregate domains and consumer-aligned domains can include all foreign data that are relevant for their consumers' use cases.

-- Jochen

**Comment by jochenchrist** [comment]
Added to FAQ: https://www.datamesh-architecture.com/#data-product-what-data-to-include-in-a-data-product-should-a-dat

**Reply by kchau158** [reply]
Hello - I concur that any aggregates that are uniquely built and published data for analytics can become a “golden source” which implies there’s ownership & accountability dimension of data product (i.e. discoverable, trustworthy, interoperable, secured, etc).  Golden source can incur a lot of cycles debating due to its operational data root and commonly associated with SSoT. I’d use something such as “golden data set” instead. 

Whatever we want to call it, I feel strongly , golden data is not something a domain team can claim right off the bat. It comes after its data product has been proven useful and provided reliable service level. It’s possible to have another future domain team that aggregates same data with additional data points and provide better services, is it not? The happy consumers and widespread use is the indicator to crown some data set (probably via the guild) as go-to golden data.

To facilitate such economic there needs to be continuous improvement and feedback capability to help producers “serve” better data products.

Thoughts? 🙂
